src/raw.js

```diff
@@ -136,6 +140,10 @@ function replaceKeyBindings(raw) {
     return full.replace(key, '?')
   })
 
+  emptyBindings.map(function(binding) {
+    sql = sql.replace(new RegExp(` ${binding}|${binding} |${binding}`, 'g'), '');
```

Comment:
@wubuzz probably the best choice here would be just to replace named binding with ? and if value is not found from bindings, then it would do the same that giving undefined to positional binding:

```js
require('knex')({client: 'pg'}).raw('?', [undefined]).toString()
// 'DEFAULT'
require('knex')({client: 'pg'}).raw('?').toString()
// '?'
```

So e.g.

```js
require('knex')({client: 'pg'}).raw(':binding').toString()
// ':binding'

require('knex')({client: 'pg'}).raw(':binding', []).toString()
// should throw an error, since named bindings require an object 

require('knex')({client: 'pg'}).raw(':binding', {}).toString()
// 'DEFAULT'

require('knex')({client: 'pg'}).raw(':binding', {binding: 'tadaa'}).toString()
// '\'tadaa\''
```
This way functionality would be consistent with positional bindings. To me current functionality is really inconsistent.

How does this sound?